music21/scale/__init__.py

```diff
@@ -842,58 +857,6 @@ def buildNetwork(self, mode=None):
         # might also set weights for tonic and dominant here
 
 
-class AbstractHarmonicMinorScale(AbstractScale):
-    '''
-    A true bi-directional scale that with the augmented
```

Comment:
We're losing two documented classes without a deprecation process.  Why are these being removed?

Reply:
My thought was that it would be better to be able to create non-natural minor keys in one line, especially considering the fact that in the existing documentation, abstract scales are described as "rarely created or manipulated directly by most users." For example, with the changes in this most recent commit, in order to create a harmonic minor key, users would type:
`minorKey = key.Key(mode='harmonic minor')`
vs. what was previously required and seems unnecessary:
`minorKey = key.Key(mode='minor')`
`minorKey.abstract = AbstractHarmonicMinor()`
Additionally, as far as I can tell, no functionality that a user would manipulate is lost by moving the abstract harmonic/melodic/jazz minor scale classes entirely into the AbstractDiatonicScale class. However, this is my first PR _ever_; would it be better to leave those abstract classes?

Reply:
Smaller PRs are usually better for this very reason. This started as a bugfix and then added a proposal for a refactor. But that's two different motivations, and the bugfix probably could have been merged already were it isolated.

We would ordinarily wait a whole version to deprecate documented classes, with notice on the list and issuing warnings in the code for people still using the old pattern. For such an unclear benefit I'm -1. You can certainly wait to hear from Michael before changing the PR, though.

Thanks for the PR and welcome aboard 👍🏻

Reply:
hiya @Phantasmogenesis, absolutely no rush here or anything, and I hope I haven't sounded short, I am genuinely pleased to see more folks lend a hand to m21. It's what I was hoping for when I marked this as a good first issue!

So please do say if you're waiting on anyone to offer guidance or direct any part of the implementation. All the best, and thanks again, it will be great to get JazzMinor and the bugfix you identified into the system. 🥇